src/hotspot/cpu/x86/stubGenerator_x86_64.cpp

```diff
@@ -5286,155 +5286,84 @@ address generate_avx_ghash_processBlocks() {
-    __ emit_data64(0x0000005200000051, relocInfo::none);
-    __ emit_data64(0x0000005400000053, relocInfo::none);
-    __ emit_data64(0x0000005600000055, relocInfo::none);
-    __ emit_data64(0x0000005800000057, relocInfo::none);
-    __ emit_data64(0x0000005a00000059, relocInfo::none);
-    __ emit_data64(0x0000006200000061, relocInfo::none);
-    __ emit_data64(0x0000006400000063, relocInfo::none);
-    __ emit_data64(0x0000006600000065, relocInfo::none);
-    __ emit_data64(0x0000006800000067, relocInfo::none);
-    __ emit_data64(0x0000006a00000069, relocInfo::none);
-    __ emit_data64(0x0000006c0000006b, relocInfo::none);
-    __ emit_data64(0x0000006e0000006d, relocInfo::none);
-    __ emit_data64(0x000000700000006f, relocInfo::none);
-    __ emit_data64(0x0000007200000071, relocInfo::none);
-    __ emit_data64(0x0000007400000073, relocInfo::none);
-    __ emit_data64(0x0000007600000075, relocInfo::none);
-    __ emit_data64(0x0000007800000077, relocInfo::none);
-    __ emit_data64(0x0000007a00000079, relocInfo::none);
-    __ emit_data64(0x0000003100000030, relocInfo::none);
-    __ emit_data64(0x0000003300000032, relocInfo::none);
-    __ emit_data64(0x0000003500000034, relocInfo::none);
-    __ emit_data64(0x0000003700000036, relocInfo::none);
-    __ emit_data64(0x0000003900000038, relocInfo::none);
-    __ emit_data64(0x0000005f0000002d, relocInfo::none);
-
-    return start;
-  }
-
-  address base64_bswap_mask_addr() {
-    __ align(CodeEntryAlignment);
-    StubCodeMark mark(this, "StubRoutines", "bswap_mask_base64");
+  address base64_shuffle_addr() {
+    __ align(64, (unsigned long) __ pc());
+    StubCodeMark mark(this, "StubRoutines", "shuffle_base64");
     address start = __ pc();
-    __ emit_data64(0x0504038002010080, relocInfo::none);
-    __ emit_data64(0x0b0a098008070680, relocInfo::none);
-    __ emit_data64(0x0908078006050480, relocInfo::none);
-    __ emit_data64(0x0f0e0d800c0b0a80, relocInfo::none);
-    __ emit_data64(0x0605048003020180, relocInfo::none);
-    __ emit_data64(0x0c0b0a8009080780, relocInfo::none);
-    __ emit_data64(0x0504038002010080, relocInfo::none);
-    __ emit_data64(0x0b0a098008070680, relocInfo::none);
-
+    assert(((unsigned long)start & 0x3f) == 0, "Alignment problem (0x%08lx)", (unsigned long)start);
+    __ emit_data64(0x0405030401020001, relocInfo::none);
+    __ emit_data64(0x0a0b090a07080607, relocInfo::none);
+    __ emit_data64(0x10110f100d0e0c0d, relocInfo::none);
+    __ emit_data64(0x1617151613141213, relocInfo::none);
+    __ emit_data64(0x1c1d1b1c191a1819, relocInfo::none);
+    __ emit_data64(0x222321221f201e1f, relocInfo::none);
+    __ emit_data64(0x2829272825262425, relocInfo::none);
+    __ emit_data64(0x2e2f2d2e2b2c2a2b, relocInfo::none);
     return start;
   }
 
-  address base64_right_shift_mask_addr() {
-    __ align(CodeEntryAlignment);
-    StubCodeMark mark(this, "StubRoutines", "right_shift_mask");
+  address base64_avx2_shuffle_addr() {
+    __ align(32);
```

Comment:
Any specific reason for aligning this to 32 bytes where are others case of base64_shuffle_addr was aligned to 64 byte boundary (CL size). Are you tightly coupling alignment with read operation width for specific tables ?

Reply:
Alignment is not strictly required here.  Since align(64) is broken, as well as using unaligned loads with 256-bit registers, this will be OK.